### ISSUE TITLE
Content Style Not Being Applied Correctly

### ISSUE THREAD

**Issue by relztic** [lokiuz/redraft#39]
### Affects Version

`v0.10.0`

### User Story

When comparing the new styles stack with the current one, while both having the same amount of items, the new string gets concat with the previous one.

This leading to styles not being applied correctly when rendering the content...

**e.g.:**

```js
arraysEqual(['a'], ['b']) // true
```

**References:**

  - Implementation: https://github.com/lokiuz/redraft/blob/0.10.0/src/ContentNode.js#L29-L33
  - Bug: https://github.com/lokiuz/redraft/blob/0.10.0/src/helpers/arrayEqual.js#L10

### Conditions of Satisfaction

Comparing equality between two arrays, regardless of their size, should return the correct`boolean`.

**Comment by richtera** [comment]
Code should be
```javascript
const arraysEqual = (arr1, arr2) => {
  if (arr1.length !== arr2.length) {
    return false;
  }
  // defining for loops with airbnb config is a pain maybe should disable some rules
  // eslint-disable-next-line
  for (let i = 0; i < arr1.length; i++) {
    if (arr1[i] !== arr2[i]) {
      return false;
    }
  }
  return true;
};
```

**Comment by lokiuz** [comment]
Merged the PR, but as I said there would still be nice to add a test case for this.

**Comment by relztic** [comment]
Say we have Draft.js implemented as our primary rich-text editor.

![screen shot 2018-03-25 at 16 02 15](https://user-images.githubusercontent.com/8301142/37879393-595d981e-3046-11e8-8040-8b31f338abb7.png)

The above image can be represented as the following JSON:

```js
{
  inlineStyleRanges: [
    { length: 6, offset: 0, style: 'bgcolor-rgb(251,160,38)' },
    { length: 6, offset: 6, style: 'bgcolor-rgb(97,189,109)' },
  ],
  text: 'Lorem ipsum.',
}
```

The issue here is that these two styles are subsequent to each other, and they both have the same amount of styles.

On an effort to reduce the amount of markup generated, when deciding whether or not to reuse the same [`ContentNode`](https://github.com/lokiuz/redraft/blob/0.10.0/src/ContentNode.js#L29-L33), the [`arraysEqual`](https://github.com/lokiuz/redraft/blob/0.10.0/src/helpers/arrayEqual.js#L10) function will fail, due to the bug described above.

![screen shot 2018-03-25 at 16 04 08](https://user-images.githubusercontent.com/8301142/37879507-18570470-3048-11e8-9219-036cd66e269f.png)

Leading to the first set of rules overlapping any subsequent one with an equal amount of styles.

**Comment by richtera** [comment]
Is this PR in npm? I didn't see a new compile.

**Comment by relztic** [comment]
The fix is now on `master`, but it hasn't made it to a release yet.

**Comment by richtera** [comment]
But this is an obvious typo where the code has ```for(init; inc; cond) { }``` instead of ```for(init; cond; inc) { }```. Should we compile the code into the fork and install from there or would you be able to post this into npm soon?

**Comment by lokiuz** [comment]
I will push a new version to npm in a few hours.

**Comment by lokiuz** [comment]
Published 0.10.1 with the patch.

**Comment by richtera** [comment]
Awesome, thank you.
